test/reconnect_test.go

```diff
@@ -1141,3 +1143,373 @@ func TestAuthExpiredForceReconnect(t *testing.T) {
+	// for _, conn := range connz.Conns {
+	// 	t.Logf("-------> %+v", conn)
+	// }
+}
+
+func TestAlwaysReconnectOnAccountMaxConnectionsExceededErr(t *testing.T) {
+	conf := createConfFile(t, []byte(`
+		listen: 127.0.0.1:-1
+                server_name: A
+		accounts: {
+			ACCT1: {
+				users: [ {user: "user1", password: "pass1"} ]
+				limits: {
+					max_conn: 1
+				}
+			},
+			ACCT2: {
+				users: [ {user: "user2", password: "pass2"} ]
+				limits: {
+					max_conn: 2
+				}
+			}
+		}
+	`))
+	defer os.Remove(conf)
+
+	s, _ := RunServerWithConfig(conf)
+	defer s.Shutdown()
+
+	clientOpts := func(name string) ([]nats.Option, chan error, chan struct{}, chan struct{}) {
+		disconnectCh := make(chan error, 10)
+		reconnectCh := make(chan struct{}, 10)
+		closedCh := make(chan struct{}, 10)
+		return []nats.Option{
+			nats.MaxReconnects(-1),
+			nats.ReconnectWait(200 * time.Millisecond),
+			nats.Timeout(200 * time.Millisecond),
+			nats.DisconnectErrHandler(func(c *nats.Conn, err error) {
+				clientID, _ := c.GetClientID()
+				t.Logf("[%v] Disconnected: %v: %v", name, clientID, err)
+				if err != nil {
+					disconnectCh <- err
+				}
+			}),
+			nats.ReconnectHandler(func(c *nats.Conn) {
+				clientID, _ := c.GetClientID()
+				t.Logf("[%v] Reconnected: %v", name, clientID)
+				reconnectCh <- struct{}{}
+			}),
+			nats.ClosedHandler(func(c *nats.Conn) {
+				clientID, _ := c.GetClientID()
+				t.Logf("[%v] Closed: %v", name, clientID)
+				closedCh <- struct{}{}
+			}),
+			nats.Name(name),
+		}, disconnectCh, reconnectCh, closedCh
+	}
+
+	opts1, _, _, _ := clientOpts("ACCT1:C:1")
+	opts1 = append(opts1, nats.UserInfo("user1", "pass1"))
+	nc1, err := nats.Connect(s.ClientURL(), opts1...)
+	if err != nil {
+		t.Fatalf("Failed to connect first client: %v", err)
+	}
+	defer nc1.Close()
+
+	if s.NumClients() != 1 {
+		t.Fatalf("Expected 1 client connection to server. Got %d", s.NumClients())
+	}
+
+	// New connection fails due to limit.
+	opts2, _, _, _ := clientOpts("ACCT1:C:2")
+	opts2 = append(opts2, nats.UserInfo("user1", "pass1"))
+	_, err = nats.Connect(s.ClientURL(), opts2...)
+	if err == nil {
+		t.Fatal("Expected connection to fail due to account max connections limit")
+	}
+	if !strings.Contains(err.Error(), `maximum account active connections exceeded`) {
+		t.Fatalf("Expected second connection to be properly rejected: %v", err)
+	}
+
+	// Update account limits.
+	err = os.WriteFile(conf, []byte(`
+		listen: 127.0.0.1:-1
+                server_name: A
+		accounts: {
+			ACCT1: {
+				users: [ {user: "user1", password: "pass1"} ]
+				limits: {
+					max_conn: 10
+				}
+			},
+			ACCT2: {
+				users: [ {user: "user2", password: "pass2"} ]
+				limits: {
+					max_conn: 2
+				}
+			}
+		}
+	`), 0666)
+	if err != nil {
+		t.Fatalf("Failed to update config file: %v", err)
+	}
+
+	t.Logf("FIRST")
+	err = s.Reload()
+	if err != nil {
+		t.Fatalf("Unexpected error reloading config [%s]", err)
```

Comment:
many error fatals and logs are inconsistent in formatting (with `%v`, with `%s`, or in brackets `[%s]`)